### PR TITLE
Improved the abbreviation handling over atom sets, this is useful for…

### PR DESCRIPTION
**Pull request by johnmay**
… reactions where we don't want to abbreviate across a bond which is broken/made. The semantics are a little hard to follow but essentially we either reject the sgroup out right, or tentatively accept it (may be fixable). As in the test case, to allow -Ac and NOT -OAc we reject out right and don't mark the atoms as visited.

### CODE REVIEW

**Comment by johnmay** [comment]
hold

**Comment by johnmay** [comment]
All good now

**Comment by egonw** [comment]
@johnmay, when a PR is ready for me to review, plz just GH to invite me as reviewer.

**Review by egonw — APPROVED** [review]
